### ISSUE TITLE
Document how to continue working on a ticket with existing branch after the migration

### ISSUE THREAD

**Issue by mkoeppe** [sagemath/trac-to-github#68]
https://github.com/sagemath/trac-to-github/blob/master/docs/Migration-Trac-to-Github.md#proposed-workflow-on-github-with-transition-guide-from-trac

**Comment by kwankyu** [comment]
What is the how-to? 

I suppose the ticket is migrated as an "issue" linked to a branch in sagetrac-mirror. Then the author can use the branch for a new PR for the issue. Is this that how-to?

**Comment by mkoeppe** [comment]
Yes, I think that's it

**Comment by tobiasdiez** [comment]
But according to https://github.com/sagemath/trac-to-github/issues/73, the branches in sagemath/sagetrac-mirror are then readonly, right? So one would need to first create a new copy of the sagetrac-mirror branch it one's own fork and create then a PR from the fork. Or do I miss something?

Alternatively, we could give everyone write-access to sagetrac-mirror.

**Comment by mkoeppe** [comment]
> But according to https://github.com/sagemath/trac-to-github/issues/73, the branches in sagemath/sagetrac-mirror are then readonly, right? So one would need to first create a new copy of the sagetrac-mirror branch it one's own fork and create then a PR from the fork.

You are right; we had this discussion before. 

sagetrac-mirror must be archived so that we don't compromise the links to branches and commits that we create in the migration of the tickets.

**Comment by mkoeppe** [comment]
I've added instructions to set up a `trac` remote for the purpose of fetching old branches in 6e55223

**Comment by soehms** [comment]
What is left to do here?

**Comment by mkoeppe** [comment]
Need a bullet point on the same level as "For contributing a change that addresses an existing open Issue" in https://github.com/sagemath/trac-to-github/blob/master/docs/Migration-Trac-to-Github.md that explains the procedure

**Comment by soehms** [comment]
> Need a bullet point on the same level as "For contributing a change that addresses an existing open Issue" in https://github.com/sagemath/trac-to-github/blob/master/docs/Migration-Trac-to-Github.md that explains the procedure

I see! I make a proposal in PR #151.

It is still not clear to me how collaboration on an issue or PR will work. With Trac we use to pull the (user or public) branch shown on the ticket header, modify it, push it back to ones own user or public branch and modify the branch field accordingly. If we want a similar workflow on GitHub the user who created the PR (resp. the first PR of an issue) needs to invite others as collaborator to his fork.

If not, we will get multiple PRs on one issue as it happened for example with issue #18 in this repository. But for Sage development we will have much longer cycles for a PR to be merged. So it can happen that there are more than one PRs for an issue (such that the current one corrects the preceding one, as we do on Trac when we changed the branch field). How shall we treat this situation? How should the preceding PR be marked such that it can be dismissed afterwards?

**Comment by mkoeppe** [comment]
> With Trac we use to pull the (user or public) branch shown on the ticket header, modify it, push it back to ones own user or public branch and modify the branch field accordingly. If we want a similar workflow on GitHub the user who created the PR (resp. the first PR of an issue) needs to invite others as collaborator to his fork.

It depends on what permissions we set up on GitHub (#85). Developers who have sufficient permissions will be able to modify a pull request by pushing to the PR branch, unless the original author of the PR has unchecked the "Allow edits by maintainers" checkbox.

**Comment by mkoeppe** [comment]
> it can happen that there are more than one PRs for an issue (such that the current one corrects the preceding one, as we do on Trac when we changed the branch field). How shall we treat this situation? How should the preceding PR be marked such that it can be dismissed afterwards?

I think we can just mark obsolete PRs with the label `duplicate` and then "Close as not planned".

**Comment by tobiasdiez** [comment]
You can always open a PR against the branch in the fork. So if person A creates a PR at `sagemath/sagemath` from his fork `A/sagemath/new_feature`, then person B can create a PR from his fork `B/sagemath/new_feature` against `A/sagemath/new_feature`. Once this PR from person B is merged by person A, the changes are automatically reflected in the original PR at the main sagemath repo. 

This is handy for larger additions. For smaller fixes, you can provide "suggestions" during a PR review: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request

**Comment by soehms** [comment]
> > it can happen that there are more than one PRs for an issue (such that the current one corrects the preceding one, as we do on Trac when we changed the branch field). How shall we treat this situation? How should the preceding PR be marked such that it can be dismissed afterwards?
> 
> I think we can just mark obsolete PRs with the label `duplicate` and then "Close as not planned".

Thanks for merging and your explanations!

**Comment by soehms** [comment]
> You can always open a PR against the branch in the fork. So if person A creates a PR at `sagemath/sagemath` from his fork `A/sagemath/new_feature`, then person B can create a PR from his fork `B/sagemath/new_feature` against `A/sagemath/new_feature`. Once this PR from person B is merged by person A, the changes are automatically reflected in the original PR at the main sagemath repo.
> 
> This is handy for larger additions. For smaller fixes, you can provide "suggestions" during a PR review: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request

My impression was that this is not possible. But if it is, I think it should be the preferred workflow, since it also guarantees that each new PR builds on top of the previous one and that the author of the former has to accept it. Will PR of person B be visible on the original issue resp. PR on `sagemath/sagemath`?

I suggest adding according instructions to the document.

**Comment by tobiasdiez** [comment]
> Will PR of person B be visible on the original issue resp. PR on sagemath/sagemath?

Yes, they will appear as commits by person B in the original PR. It's the same outcome as pushing directly to `A/sagemath/new_feature`. If you then merge the original PR by person A, person B will be added as coauthor (at least if you merge using github's ui).

**Comment by kwankyu** [comment]
We are done with this issue? @soehms

**Comment by soehms** [comment]
> We are done with this issue? @soehms

From my point of view, yes! But since Matthias changed the milestone three days ago, he may have something in mind.

**Comment by mkoeppe** [comment]
No, I think it's done; I just wanted to clear that milestone so that I could close it.
